scripts/secnetperf-helpers.psm1

```diff
@@ -63,117 +127,133 @@ function Stop-RemoteServer {
     Stop-Job -Job $Job
     $RemoteResult = Receive-Job -Job $Job -ErrorAction Stop
     $RemoteResult = $RemoteResult -join "`n"
-    Write-GHWarning $RemoteResult.ToString()
+    Write-Host $RemoteResult.ToString()
     throw "Server failed to stop!"
 }
 
-# Invokes all the secnetperf tests.
-function Invoke-SecnetperfTest($MsQuicCommit, $commands, $exe, $start, $LogProfile) {
-
-    Write-Host "Running Secnetperf tests..."
-
-    $SQL = @"
-"@
-    $json = @{}
+class TestResult {
+    [String]$Metric;
+    [System.Object[]]$Values;
+    [Boolean]$HasFailures;
+
+    TestResult (
+        [String]$Metric,
+        [System.Object[]]$Values,
+        [Boolean]$HasFailures
+    ) {
+        $this.Metric = $Metric;
+        $this.Values = $Values;
+        $this.HasFailures = $HasFailures;
+    }
+}
 
+# Invokes secnetperf with the given arguments for both TCP and QUIC.
+function Invoke-Secnetperf {
+    param ($Session, $RemoteName, $RemoteDir, $SecNetPerfPath, $LogProfile, $ExeArgs)
+
+    $values = @(@(), @())
+    $hasFailures = $false
+
+    # TODO: This logic is pretty fragile. Needs improvement.
+    $metric = "throughput-download"
+    if ($exeArgs.Contains("plat:1")) {
+        $metric = "latency"
+    } elseif ($exeArgs.Contains("prate:1")) {
+        $metric = "hps"
```

Comment:
How are these "plat" and "prate" associated with each metric name? Looks really high context naming

Reply:
As mentioned in the comment above, this stuff is fragile and needs to be improved in future PRs.